sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala

```diff
@@ -39,27 +39,36 @@ object StringUtils extends Logging {
    * throw an [[AnalysisException]].
    *
    * @param pattern the SQL pattern to convert
+   * @param escapeStr the escape string contains one character.
    * @return the equivalent Java regular expression of the pattern
    */
-  def escapeLikeRegex(pattern: String): String = {
+  def escapeLikeRegex(pattern: String, escapeStr: String): String = {
+    val escapeChar = escapeStr.charAt(0)
```

Comment:
this might need to `assert(escapeStr.length == 1)`?

Reply:
OK. Thanks! A good idea.